core/src/main/java/org/infinispan/factories/DataContainerFactory.java

```diff
@@ -65,7 +68,8 @@ public Object construct(String componentName) {
                int addressCount = memoryConfiguration.addressCount();
                if (shouldSegment) {
                   int segments = clusteringConfiguration.hash().numSegments();
-                  Supplier mapSupplier = () -> createAndStartOffHeapConcurrentMap(addressCount, segments);
+                  Supplier<PeekableTouchableMap<WrappedBytes, InternalCacheEntry<WrappedBytes, WrappedBytes>>>
+                        mapSupplier = () -> createAndStartOffHeapConcurrentMap(addressCount, segments);
```

Comment:
:+1: 

For master, the declaration would be simpler if you made `PeekableTouchableMap<K, V>` implement `ConcurrentMap<K, InternalCacheEntry<K, V>>`, and the `PeekableMap` and `TouchableMap` interfaces also don't seem to be used by themselves:

https://github.com/infinispan/infinispan/commit/1f21b704cc226973d0ac243013bd2b0487ab7003

I'd go even further and change the interface name, e.g. `DataContainerSegment`, so there's no confusion because the `V` means something else compared to `Map`.

Reply:
Yeah part of the problem is the disparity between 9.4.x and master. I can try to consolidate them on master better.

Reply:
Yeah one of my issues was that PeekableMap is in commons, I wasn't quite sure about removing it.

Reply:
We don't need to remove it immediately, but don't have to keep implementing it either, because the per-segment maps were never exposed.

Reply:
Sure, I can just deprecate it. More than likely no one is using it anyways.

Reply:
I will do that in master only though.